examples/gitlab/gitlab-push-listener.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: gitlab-echo-template
+spec:
+  params:
+  - name: gitrevision
+  - name: gitrepositoryurl
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: TaskRun
+    metadata:
+      generateName: gitlab-run-
+    spec:
+      taskSpec:
+        inputs:
+          resources:
+            - name: source
+              type: git
+        steps:
+          - image: ubuntu
+            script: |
+              #! /bin/bash
+              ls -al $(inputs.resources.source.path)
+      inputs:
+        resources:
+        - name: source
+          resourceSpec:
+            type: git
+            params:
+            - name: revision
+              value: $(params.gitrevision)
+            - name: url
+              value: $(params.gitrepositoryurl)
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: gitlab-push-binding
+spec:
+  params:
+  - name: gitrevision
+    value: $(body.checkout_sha)
+  - name: gitrepositoryurl
+    value: $(body.repository.git_http_url)
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: gitlab-secret
+type: Opaque
+stringData:
+  gitlabToken: abcde
+---
+apiVersion: tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: gitlab-listener
+spec:
+  # from examples/role-resources/servicaccount.yaml
+  serviceAccountName: tekton-triggers-example-sa
```

Comment:
I think we need to have the ServiceAccount `tekton-triggers-example-sa` in this file.

I think we'll also need a Role & RoleBinding, because the idea is that anyone can just `kubectl apply -f` this file, and it'll work, right?

Reply:
Added the bit about applying role resources a a pre-requisite